internal/store/store.go

```diff
@@ -794,6 +833,47 @@ func (s Store) GetIngressClassV1(name string) (*networkingv1.IngressClass, error
 	return p.(*networkingv1.IngressClass), nil
 }
 
+// GetIngressClassV1 returns the 'name' IngressClass resource
+func (s Store) GetIngressClassParametersV1Alpha1() (*kongv1alpha1.IngressClassParameters, error) {
```

Comment:
We may want to make this accept an IngressClass name and have the parser provide the class it wants (probably by making `store.ingressClass` a public variable. Ostensibly we want to support multiple classes in the future, so we'd need to retrieve several params. That could easily be deferred to whenever we actually do it though--I don't think we'd use this function outside the one location here and it shouldn't be hard to adapt usage to a revised signature.

Reply:
Created an issue for this in: #2705. I'd prefer not to touch this one as this PR got big enough in my eyes already.